InstantSearch.js/routing-seo-friendly/src/search-router.js

```diff
@@ -24,13 +24,19 @@ const router = instantsearch.routers.history({
     const pathname = routeState.category
       ? `/${getCategorySlug(routeState.category)}/`
       : '';
+    const queryParameters = {};
 
-    const searchState = {};
-    if (routeState.q) searchState.q = encodeURIComponent(routeState.q);
-    if (routeState.p !== 1) searchState.p = routeState.p;
-    if (routeState.brands) searchState.brands = routeState.brands;
+    if (routeState.q) {
+      queryParameters.q = encodeURIComponent(routeState.q);
+    }
+    if (routeState.p !== 1) {
+      queryParameters.p = routeState.p;
+    }
+    if (routeState.brands) {
+      queryParameters.brands = encodeURIComponent(routeState.brands);
```

Comment:
You have to decode it otherwise the sync is not working anymore. Note that we also encode the delimiter `|` not sure it's the expected behaviour. We might want to only encode the value but not the delimiter.

Reply:
We're now repeating the array values:

```
?brands=Metra&brands=Samsung
```